web-admin/src/features/projects/status/WaitForDeployment.ts

```diff
@@ -0,0 +1,192 @@
+import {
+  createAdminServiceGetProject,
+  V1DeploymentStatus,
+  type V1GetProjectResponse,
+} from "@rilldata/web-admin/client";
+import { baseGetProjectQueryOptions } from "@rilldata/web-admin/features/projects/status/selectors";
+import type { CompoundQueryResult } from "@rilldata/web-common/features/compound-query-result";
+import {
+  fetchResources,
+  ResourceKind,
+  SingletonProjectParserName,
+  useResource,
+} from "@rilldata/web-common/features/entity-management/resource-selectors";
+import { eventBus } from "@rilldata/web-common/lib/event-bus/event-bus";
+import { queryClient } from "@rilldata/web-common/lib/svelte-query/globalQueryClient";
+import { V1ReconcileStatus } from "@rilldata/web-common/runtime-client";
+import { derived, type Unsubscriber } from "svelte/store";
+
+const PollTime = 1000;
+
+export class WaitForDeployment {
+  private errored = false;
+
+  // The user is 1st redirected to `invite` page. They might take some time on there or be very quick.
+  // We use this boolean to identify if a deployment already succeeded by the time user lands on dashboards page.
+  private shouldRedirect = false;
+
+  private readonly unsub: Unsubscriber;
+
+  private static instance: WaitForDeployment | undefined;
```

Comment:
Given `instance` has an important meaning throughout our app (especially regarding deployments), maybe `singleton` would be a better name